pkg/event/target/queuestore.go

```diff
@@ -0,0 +1,240 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package target
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/minio/minio/pkg/event"
+)
+
+const (
+	maxLimit = 10000 // Max store limit.
+	eventExt = ".event"
+)
+
+type QueueStore struct {
+	sync.RWMutex
+	directory string
+	eC        uint16
+	limit     uint16
+}
+
+// NewQueueStore - Filestore for persisting events.
+func NewQueueStore(directory string, limit uint16) *QueueStore {
+	if limit == 0 {
+		limit = maxLimit
+	}
+	queueStore := &QueueStore{
+		directory: directory,
+		limit:     limit,
+	}
+	return queueStore
+}
+
+// Staging directory.
+func tmpPath(directory string) string {
+	return directory + "/tmp"
+}
+
+// Open - Creates the directory if not present.
+func (store *QueueStore) Open() error {
+	store.Lock()
+	defer store.Unlock()
+
+	if err := store.resetTmp(); err != nil {
+		return err
+	}
+
+	if _, err := os.Stat(store.directory); err != nil {
+		if os.IsNotExist(err) {
+			if merr := os.MkdirAll(store.directory, os.FileMode(0770)); merr != nil {
+				return merr
+			}
+		} else {
+			return err
+		}
+	}
+
+	tmpDir := tmpPath(store.directory)
+	if terr := os.MkdirAll(tmpDir, os.FileMode(0770)); terr != nil {
```

Comment:
You only need this os.MkdirAll() will create all parents including the store.directory, so previous `os.Stat()` are all not useful.

Reply:
Yes correct. Done.